version: '3.8'

services:
  auth-service:
    build: ./auth_service
    depends_on:
      - postgres
    networks:
      - app-network
  api-gateway:
    image: nginx:latest
    volumes:
      - ./api_gateway/nginx.conf:/etc/nginx/nginx.conf:ro
    ports:
      - "8081:8081"
    depends_on:
      - auth-service
    networks:
      - app-network
  postgres:
    image: postgres:latest
    restart: always
    env_file:
      - ./auth_service/credentials/.env
    ports:
      - "5432:5432"
    volumes:
      - ./postgresql_storage:/var/lib/postgresql/data
    networks:
      - app-network
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U admin -d auth" ]
      interval: 10s
      retries: 3
      start_period: 5s
      timeout: 5s
networks:
  app-network:
    driver: bridge
